File: C#/Quality-Programing-Code/11.Unit-Testing/SchoolSystem/School.cs
namespace SchoolSystem
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;
    using System.Threading.Tasks;

    public class School
    {
        private string name;

        public School(string name)
        {
            this.Name = name;
            this.Courses = new List<Course>();
        }

        public string Name
        {
            get 
            { 
                return this.name; 
            }

            set 
            {
                if (string.IsNullOrEmpty(value))
                {
                    throw new ArgumentException("Name of the school cannot be null or empty");
                }

                this.name = value; 
            }
        }
        
        public List<Course> Courses { get; set; }

        public void AddCourse(Course course) 
        {
            bool courseExists = this.CheckIfCourseExists(course);

            if (courseExists) 
            {
                throw new ArgumentException("Course already exists");
            }

            this.Courses.Add(course);
        }

        public void RemoveCourse(Course course)
        {
            bool courseExists = this.CheckIfCourseExists(course);

            if (!courseExists)
            {
                throw new ArgumentException("Cannot remove non-existent course");
            }

            this.Courses.Remove(course);
        }

        /*
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.AppendLine("School:" + this.Name);

            foreach (var course in this.Courses)
            {
                sb.AppendLine(course.ToString());
            }

            return sb.ToString();
        }
        */

        private bool CheckIfCourseExists(Course course)
        {
            var result = this.Courses.Find(x => x.Name == course.Name);
            var isFound = false;

            if (result != null)
            {
                isFound = true;
            }

            return isFound;
        }
    }
}

File: Javascript/Javascript-APIs/1.Javascript-OOP/Classic-OOP/School.js
var School = {
	init: function(name,town,setOfClasses){
		this.name= name;
		this.town = town;
		if(!setOfClasses){
			this.setOfClasses = [];
		}else{
			this.setOfClasses = setOfClasses;
		}
	},
	addClass: function(newClass){
		this.setOfClasses.push(newClass);
	},
	toString: function(){
		var result = "";
		result+= "Name:"+this.name + " ;Town:" + this.town + "\n";
		for(var i=0;i<this.setOfClasses.length;i+=1){
			result += this.setOfClasses[i].toString();
		}

		return result;
	}
}